private/protocol/jsonrpc/unmarshal_error.go

```diff
@@ -13,17 +13,37 @@ import (
 	"github.com/aws/aws-sdk-go/private/protocol/json/jsonutil"
 )
 
+const (
+	awsQueryError = "x-amzn-query-error"
+)
+
 // UnmarshalTypedError provides unmarshaling errors API response errors
 // for both typed and untyped errors.
 type UnmarshalTypedError struct {
 	exceptions map[string]func(protocol.ResponseMetadata) error
+	parseQueryError bool
 }
 
 // NewUnmarshalTypedError returns an UnmarshalTypedError initialized for the
 // set of exception names to the error unmarshalers
 func NewUnmarshalTypedError(exceptions map[string]func(protocol.ResponseMetadata) error) *UnmarshalTypedError {
 	return &UnmarshalTypedError{
 		exceptions: exceptions,
+		parseQueryError: false,
+	}
+}
+
+func NewUnmarshalTypedErrorWithOptions(exceptions map[string]func(protocol.ResponseMetadata) error, optFns ...func(*UnmarshalTypedError)) *UnmarshalTypedError {
+	err := NewUnmarshalTypedError(exceptions)
+	if len(optFns) == 1 {
+		optFns[0](err)
+	}
+	return err
```

Comment:
Recommend changing this to:
```go
unmarshaler := NewUnmarshalTypedError(exceptions)
for _, fn := range optFns {
    fn(unmarshaler)
}
```

Mainly to avoid frustrating bugs in the future (not looping the options), but also to not use a variable name common for actual errors.

Reply:
Probably unrelated to the change.